packages/wmr/src/lib/prerender.js

```diff
@@ -96,8 +96,33 @@ async function workerCode({ cwd, out, publicPath, customRoutes }) {
 	let head = { lang: '', title: '', elements: new Set() };
 	globalThis.wmr = { ssr: { head } };
 
+	// Define `_fetch` to avoid infinite loop in globalThis.fetch.
+	let _fetch;
+	const nodeVersion = process.version;
+	const isNodeVersionOver18 = Number(nodeVersion.split('.')[0].slice(1)) >= 18;
+	// Native fetch is only available in Node.js 18 later.
+	if (isNodeVersionOver18) {
+		_fetch = fetch;
+	}
+	// @ts-ignore
+	delete globalThis.fetch;
 	// @ts-ignore
-	globalThis.fetch = async url => {
+	globalThis.fetch = async (url, options) => {
```

Comment:
I don't think we want to be doing this, as it breaks user polyfills. My fault as I admittedly missed it in the previous PR.

Maybe if Node < 18, use the user defined `fetch()` if it exists, else, reassign to our "polyfill". If Node >= 18, we'll need to reassign in order to inject our file reader.

What do you think?

Reply:
Well, now that you mention it, it makes sense.
I very much agree that it is not a good idea to override our implementation by removing what the user has defined.

We felt it appropriate to leave the implementation of `fetch` itself to the user, with our concerns limited to whether `fetch` exists and whether Node.js version 18 or higher is used.

Reply:
Given the specific code, I feel it can be simplified to this point.
```js
if (globalThis.fetch === undefined) {
    // @ts-ignore
    globalThis.fetch = async url => {
		const pattern = /^\//;
		const isLocalFile = pattern.test(String(url));
		if (isLocalFile) {
			const text = () => fs.readFile(`${out}/${String(url).replace(/^\//, '')}`, 'utf-8');
			return { text, json: () => text().then(JSON.parse) };
		}
		console.warn(`fetch is not implemented in Node.js, please upgrade to Node.js 18 or above`);
	};
}
```

Reply:
Yep, I think that looks good for a Node < v18 impl.

Not sure what to do about v18 though. We either override `globalThis.fetch` to add automatic local file handling or we allow the user to override it themselves. No way to do both, I don't think. Hmm.

Reply:
I am also very confused.
I have the same implementation on my personal blog, but my motivation for creating this PR was to implement this part on the wmr side so that the user does not have to implement it!
https://github.com/takurinton/blog.takurinton.dev/blob/main/public/prerender.tsx#L9-L12

So,  
> We either override globalThis.fetch to add automatic local file handling

I feel this is a good idea. I cannot be certain that this is always the right one....

Reply:
Thought about it a bit more and I think for Node 18 we should just override `globalThis.fetch` with our own implementation.

While yes, this won't allow users to provide their own, there should be few, if any, reasons why anyone would want to do this, and I can think of none where they'd _need_ to.